content/chapters/data/lab/support/memory-security/buff_leak.c

```diff
@@ -0,0 +1,84 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+int a = 9;
+int b;
+
+int main()
+{
+    int c = 15;
+    int *k = malloc(sizeof(int));
+    *k = 100;
+    char d;
+    char *p = &d;
+
+    long long int offset;
+
+    printf("&a = %lld\n",(long long int)&a);
+    printf("&b = %lld\n",(long long int)&b);
+    printf("&c = %lld\n",(long long int)&c);
+    printf("k = %lld\n", (long long int)k);
+    printf("The player pointer is %lld\n", (long long int)p);
+
+Lagain1:
+    // insert offset of p to c
+    printf("What's the offset of c from p?\n");
+    scanf("%lld", &offset);
+
+    if((int*)(p + offset) == &c)
+    {
+        printf("Congrats, you have leaked: %d\n\n", c);
+    }
+    else
+    {
```

Comment:
```suggestion
    if ((int*)(p + offset) == &c)
        printf("Congrats, you have leaked: %d\n\n", c);
    else {
```

Reply:
I think that using no braces on one arm and braces on the other is a sign of bad taste.

Reply:
But, I see that you like the weird way of putting braces.

Reply:
I'll make a final linter pass after the lab is done.

Reply:
For C we use the Linux kernel coding style [1]. That's why.